### PR TITLE
fix getPath method

### PR DESCRIPTION
**Pull request by ex37**
Reuse `getPath` function instead of duplicating code.
Within `getPath` pass `this.path` to  `pathWithParams`, cause `this.currentPath`, if redeclared, will most likely return a string with already resolved params.

### CODE REVIEW

File: projects/akita-ng-fire/src/lib/collection/collection.service.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class CollectionService<S extends EntityState<any, string>>  {
   }
 
   /** The path to the collection in Firestore */
-  get path(): string | Observable<string> {
```
Comment (ex37) [inline projects/akita-ng-fire/src/lib/collection/collection.service.ts]: @GrandSchtroumpf ok so it turns out that removing this line causes type errors also Subcollection service. So maybe it's worth removing SubcollectionService at this point, anyway it's not compatible to current CollectionService code.

Comment (ex37) [reply projects/akita-ng-fire/src/lib/collection/collection.service.ts]: And I can write an util function, like we discussed before, using same logic found in SubcollectionService

Comment (GrandSchtroumpf) [reply projects/akita-ng-fire/src/lib/collection/collection.service.ts]: I would like not to introduce too much breaking changes for next version. Please consider submitting a PR that doesn't breaks `Subcollection`.

Comment (ex37) [reply projects/akita-ng-fire/src/lib/collection/collection.service.ts]: Sure that's possible, but as we already discussed, the SubcollectionService is not working correctly with current v2, and leaving it as deprecated will mean it's working, just not recommended  to use, which is confusing, cause it's not. So either you need to rewrite it for backwards compatibility, which probably isn't worth it, or just remove by introducing a breaking change, but we also can document how to easily migrate old code, by introducing util functions.

Comment (GrandSchtroumpf) [reply projects/akita-ng-fire/src/lib/collection/collection.service.ts]: Ok, let's do that. Can you update the Breaking change file at the root of the project to explain the migration ?
Thx

File: projects/akita-ng-fire/src/lib/utils/sync-with-router.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { getPathParams } from './path-with-params';
+import { CollectionService, CollectionState } from '../collection/collection.service';
+import { distinctUntilChanged, filter, switchMap, shareReplay } from 'rxjs/operators';
+import { RouterQuery } from '@datorama/akita-ng-router-store';
+import { Observable } from 'rxjs';
+import { DocumentChangeAction } from '@angular/fire/firestore';
+
+export function syncWithRouter<Service extends CollectionService<CollectionState<E>>, E>(
+  this: Service,
+  routerQuery: RouterQuery
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/utils/sync-with-router.ts]: You don't need to give the routerQuery as second parameter. You could do something like : 
```typescript
type ServiceWithRouter<E> = CollectionService<CollectionState<E>> & { routerQuery: RouterQuery };
export function syncWithRouter<Service extends ServiceWithRouter<E>, E>(
  this: Service
): Observable<DocumentChangeAction<E>[]> {}
```

The limitation doing that is that you force developer to use the naming `routerQuery`.

**Comment by ex37** [comment]
@GrandSchtroumpf  a small update on the `syncWithRouter` method. While it works pretty well, there is one thing which should probably be mentioned in the docs to make sure people end up with smooth migration. To make it fully work like before, besides adding `sync = syncWithRouter.bind(this, this.routerQuery);` one should also add smth like:
```
  get currentPath() {
    const params = this.routerQuery.getParams<string>();
    return this.getPath({ params });
  }
```
to make sure entity operations also work. What do you think?

**Comment by GrandSchtroumpf** [comment]
Hmmm I think I see your point, but if for a reason or another there is no param in the router at that time you'll end up with an infinite loop (getPath calls currentPath if there is no params). So you might consider something like : 
```typescript
return params ? this.getPath({ params }) : null;
```
I think it's important to consider this possibility as the service can be called from anywhere.
Could you make a PR for that ?

**Comment by ex37** [comment]
I see. Then, ideally, it should be something like this:
```
    const params = this.routerQuery.getParams<string>();
    const paramsAvailable = getPathParams(this.path).every(key => params[key]);
    if (!paramsAvailable) throw 'Route params required. Consider manually passing params.';
    return this.getPath({ params });
```

Cause  looking at `getPath` method
```
  protected getPath(options: PathParams) {
    return (options && options.params)
      ? pathWithParams(this.path, options.params)
      : this.currentPath;
  }
```
It calls currentPath only if params were not manually passed. And in case not all of them are found in the route, you'll end up with doing write operation in firebase with a path like `collectionname/:id/somedoc`

**Comment by GrandSchtroumpf** [comment]
Ok, would it make sense to create a utils function ? Or you think it's better just to right it down on the doc ?

**Comment by ex37** [comment]
Probably should be an util. Btw, inside the docs, we can possibly just show how to implement a custom SubCollectionService which extends the collection service, for like a super easy way to migrate.

**Comment by GrandSchtroumpf** [comment]
Yes, sounds like a good idea. Would you consider creating a PR for all that ?

**Pull request merged**